### PR TITLE
Update docs to use the correct branch name for older rails versions

### PR DESCRIPTION
**Pull request by akcrono**
Addresses #2452

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,9 +33,10 @@ Use **[`rspec-rails` 1.x][]** for Rails 2.x.
 
    # Or, run against the master branch
    # (requires master-branch versions of all related RSpec libraries)
+   # Replace branch name with the appropriate branch for your rails version
    group :development, :test do
      %w[rspec-core rspec-expectations rspec-mocks rspec-rails rspec-support].each do |lib|
-       gem lib, :git => "https://github.com/rspec/#{lib}.git", :branch => 'master'
+       gem lib, :git => "https://github.com/rspec/#{lib}.git", :branch => '3-9-maintenance'
```
Comment (pirj) [inline README.md:39]: The latest on the `3-9-maintenance` branch is released and can be fetched from RubyGems with `~> 3.9` as described above.

Comment (akcrono) [reply README.md:39]: This isn't mentioned with the block gem declaration.

**Comment by pirj** [comment]
Thank you for your contribution, but we do not maintain the previous versions.

**Comment by JonRowe** [comment]
I think updating the readme is a reasonable ask here.

File: README.md [hunk 2]
```diff
@@ -33,9 +33,10 @@ Use **[`rspec-rails` 1.x][]** for Rails 2.x.
 
    # Or, run against the master branch
    # (requires master-branch versions of all related RSpec libraries)
+   # Replace branch name with the appropriate branch for your rails version
```
Comment (JonRowe) [inline README.md:36]: This whole comment block should be:
```
# Or, run against the maintenance branch
# (requires the same maintenance branch for all related RSpec libraries)
```

Comment (JonRowe) [reply README.md:36]: Additionally prehaps we should add:

```
# Note that 3.9 is end of life and all changes have been released.
```

Comment (akcrono) [reply README.md]: Good points

**Pull request merged**